087-external-certificate-manager.md

```diff
@@ -0,0 +1,241 @@
+
+### API
+
+The existing `spec.clusterCa` and `spec.clientsCa` fields will be extended to add a new property `type`:
+
+```yaml
+spec:
+  clusterCa:
+    validityDays: <integer> # notBefore=now, notAfter=now + validityDays
+    generateCertificateAuthority: <boolean>
+    generateSecretOwnerReference: <boolean>
+    renewalDays: <integer> # days before notAfter when we should start renewal
+    certificateExpirationPolicy: <renew-certificate|replace-key>
+    type: <internal|cert-manager.io> # (1)
+    certManagerIssuerRef: # (2)
+      name: <string>
+      kind: <Issuer|ClusterIssuer>
+      group: <string> # cert-manager.io by default
+    publicCert: # (3)
+      secretName: <string>
+      certificate: <string>
+```
+
+1. The`type` property will default to `internal` when not set and will use the existing behaviour, allowing backwards compatibility.
+   The option `cert-manager.io` will only be valid if `generateCertificateAuthority` is set to `false`.
+2. The property `certManagerIssuerRef` will only be used by Strimzi if `type` is set to `cert-manager.io`.
+  The `name`, `kind`, and `group` properties will be copied over into the `Certificate` custom resource Strimzi creates.
+3. The property `publicCert` will only be used by Strimzi if `type` is set to `cert-manager.io`.
+  The `secretName` and `certificate` properties will be used to locate the CA public certificate that must be trusted by Strimzi components in order to trust the end-entity certificates that cert-manager issues.
+
+### User steps
+
+To make use of this new option the user will have to:
+
+1. Install cert-manager.
+2. Create an `Issuer` or `ClusterIssuer` custom resource.
+3. Create a `Secret` containing the CA public cert for Strimzi to trust.
+   Users can optionally use [trust-manager](https://cert-manager.io/docs/trust/trust-manager/) to create this Secret, but they are responsible for installing trust-manager and creating the `Bundle` CR with a single `sources` entry.
+4. Create a `Kafka` resource with `clusterCa.type` and/or `clientsCa.type` set to `cert-manager.io`, and `certManagerIssuerRef` and `publicCert` configured.
+
+### Internal handling of certificates
+
+Strimzi uses two generation annotations to track how Cluster CA certificates are trusted and used:
+* `strimzi.io/cluster-ca-key-generation` on Kafka pods to indicate the CA generation trusted by that pod
+* `strimzi.io/cluster-ca-cert-generation` on Secrets containing certs to indicate the CA generation that signed those certs
+
+The `strimzi.io/cluster-ca-cert-generation` annotation is also used on Kafka pods to indicate the CA generation that signed the certs it is currently presenting (i.e. the CA generation currently in use).
+
+When using cert-manager for issuing certificates Strimzi will continue to use these generations in the same way.
+Rather than directly mounting cert-manager Secrets and the user-provided CA cert Secret, Strimzi will copy the certificates into its own existing Secrets.
```

Comment:
I would remove this sentence. Before reading the next paragraph where everything is well explained, I wrote the following questions: 

> "its own existing Secrets." ... are you referring to the xxxxx-cluster-ca-cert Secret?
Will the operator have a logic which at each reconciliation (periodically) watches the cert-manager Secret and update to its own? Or how is it going to work? Also, any reasons why not mounting the cert-manager Secret? For backward compatibility? (sorry too many questions).

Because the above sentence was not clear to me. To avoid confusion I would just remove it and leaving to the next paragraph the deep explanation of how things are going to work.